tests/conftest.py

```diff
@@ -55,6 +45,111 @@ def pytest_ignore_collect(path, config):
             return True
 
 
+def pytest_configure(config):
+    config.addinivalue_line(
+        'markers',
+        'bdb(): use bigchaindb')
+
+
+@pytest.fixture(autouse=True)
+def _bdb_marker(request):
+    if request.keywords.get('bdb', None):
+        request.getfixturevalue('_bdb')
+
+
+@pytest.fixture(autouse=True)
+def _restore_config(_configure_bigchaindb):
+    from bigchaindb import config, config_utils
+    config_before_test = copy.deepcopy(config)
+    yield
+    config_utils.set_config(config_before_test)
+
+
+@pytest.fixture
+def _restore_dbs(request):
+    from bigchaindb.backend import connect, schema
+    from bigchaindb.common.exceptions import DatabaseDoesNotExist
+    from .utils import list_dbs
+    conn = connect()
+    dbs_before_test = list_dbs(conn)
+    yield
+    dbs_after_test = list_dbs(conn)
+    dbs_to_delete = (
+        db for db in set(dbs_after_test) - set(dbs_before_test)
+        if TEST_DB_NAME not in db
+    )
+    print(dbs_to_delete)
+    for db in dbs_to_delete:
+        try:
+            schema.drop_database(conn, db)
+        except DatabaseDoesNotExist:
+            pass
+
+
+@pytest.fixture(scope='session')
+def _configure_bigchaindb(request):
```

Comment:
I'm on the fence about renaming this to something else, like `_bigchaindb_configuration` or `_test_bigchaindb_configuration`. "Configure" to me sounds like it's *doing* something rather than *returning* a config, although I guess one could argue that this is, in fact, actually "configuring" via `config_utils.set_config()`.